pkg/process/config/yaml_config.go

```diff
@@ -270,3 +274,22 @@ func (a *AgentConfig) setCheckInterval(ns, check, checkKey string) {
 		a.CheckIntervals[checkKey] = time.Duration(interval) * time.Second
 	}
 }
+
+// Separate handler for initializing the process discovery check.
+// Since it has its own unique object, we need to handle loading in the check config differently separately
+// from the other checks.
+func (a *AgentConfig) initProcessDiscoveryCheck() {
+	root := key(ns, "process_discovery")
+
+	// We don't need to check if the key exists since we already bound it to a default in InitConfig
+	a.CheckIntervals[DiscoveryCheckName] = config.Datadog.GetDuration(key(root, "interval"))
+
+	// Discovery check should be only enabled when process_config.process_discovery.enabled = true and
+	// process_config.enabled is set to "false". This effectively makes sure the check only runs when other checks are
```

Comment:
`when other checks are disabled` -> `when process check is disabled`. 
If `process_config.enabled = false`, we're still running `container` and `rt_container` checks
https://docs.datadoghq.com/infrastructure/process/?tab=linuxwindows